sadu-updater/src/main/java/de/chojo/sadu/updater/SqlUpdater.java

```diff
@@ -216,7 +219,9 @@ private void performUpdate(Patch patch) throws SQLException {
             }
             conn.commit();
         } catch (SQLException e) {
-            log.warn("Database update failed", e);
+            String message = "Database update failed while executing the " + statementsOfPatchExecuted +
+                    ". statement of the patch " + patch.version() + "!";
+            log.warn(message, e);
```

Comment:
I would log the statement itself in the inner catch block of the try statement and just rethrow the exception.
That is much more helpful than starting to count on large patch files.
Also no need to log the exception twice since it is already logged.

Reply:
Resolved this in the commit ["Applied Suggested Changes"](https://github.com/rainbowdashlabs/sadu/pull/115/commits/0c6048be4f7689f0ba9dfa094842bf0471516369). If the query itself throws an exception, it is handled in the inner catch block.